pkg/storage/memory/memory_test.go

```diff
@@ -28,12 +29,33 @@ func TestStaticTupleIteratorNoRace(t *testing.T) {
 	defer iter.Stop()
 
 	go func() {
-		_, err := iter.Next()
+		_, err := iter.Next(context.Background())
 		require.NoError(t, err)
 	}()
 
 	go func() {
-		_, err := iter.Next()
+		_, err := iter.Next(context.Background())
 		require.NoError(t, err)
 	}()
 }
+
+func TestStaticTupleIteratorContextCanceled(t *testing.T) {
```

Comment:
Can you add a test `TestStaticTupleIteratorContextDeadlineExceeded`?

Reply:
Yep, added 👍